x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/rule_delete/basic_license_essentials_tier/delete_rules_bulk.ts

```diff
@@ -51,7 +55,9 @@ export default ({ getService }: FtrProviderContext): void => {
           .expect(200);
 
         const bodyToCompare = removeServerGeneratedProperties(body[0]);
-        expect(bodyToCompare).to.eql(getSimpleRuleOutput());
+        const expectedRule = updateUsername(getSimpleRuleOutput(), ELASTICSEARCH_USERNAME);
```

Comment:
Seeing all these `updateUsername`s added here doesn't feel great; I'll reiterate [@banderror's previous comment](https://github.com/elastic/kibana/pull/172173#discussion_r1424404725) pointing out that we now modify both the actual value and the expected value, which does seem a little silly. 

I really like `expect.objectContaining` as referenced, but it can definitely get a little verbose and hard to read; maybe we could hide some of that verbosity in a declarative matcher? It's not obvious how one would go about adding a matcher to kbn-expect, but we could certainly just write a function that compares the appropriate fields and ignores the rest:

```ts
assertRuleEquality({ actual: body[0], expected: getSimpleRuleOutput() })
```

Reply:
I'll look to create some follow up issues to track these suggestions!